apps/docs/content/guides/auth/server-side/sveltekit.mdx

```diff
@@ -0,0 +1,636 @@
+<StepHikeCompact.Code>
+
+<CH.Code>
+
+```ts src/routes/auth/+page.server.ts
+import { redirect } from '@sveltejs/kit'
+
+import type { Actions } from './$types'
+
+export const actions: Actions = {
+  signup: async ({ request, locals: { supabase } }) => {
+    const formData = await request.formData()
+    const email = formData.get('email') as string
+    const password = formData.get('password') as string
+
+    const { error } = await supabase.auth.signUp({ email, password })
+    if (error) {
+      console.error(error)
+      return redirect(303, '/auth/error')
+    } else {
+      return redirect(303, '/')
+    }
+  },
+  login: async ({ request, locals: { supabase } }) => {
+    const formData = await request.formData()
+    const email = formData.get('email') as string
+    const password = formData.get('password') as string
+
+    const { error } = await supabase.auth.signInWithPassword({ email, password })
+    if (error) {
+      console.error(error)
+      return redirect(303, '/auth/error')
+    } else {
+      return redirect(303, '/private')
+    }
+  },
+}
+```
+
+```svelte src/routes/auth/+page.svelte
+<form method="POST" action="?/login">
+	<label>
+		Email
+		<input name="email" type="email" />
+	</label>
+	<label>
+		Password
+		<input name="password" type="password" />
+	</label>
+	<button>Login</button>
+	<button formaction="?/signup">Sign up</button>
+</form>
+```
+
+```svelte src/routes/auth/+layout.svelte
+<header>
+	<nav>
+		<a href="/">Home</a>
+	</nav>
+</header>
+<slot />
+```
+
+```svelte src/routes/auth/error/+page.svelte
+<p>Login error</p>
+```
+
+</CH.Code>
+
+</StepHikeCompact.Code>
+
+</StepHikeCompact.Step>
+
+<StepHikeCompact.Step step={10}>
+
+<StepHikeCompact.Details title="Create the signup confirmation route">
+
+Finish the signup flow by creating the API route to handle email verification.
+
+</StepHikeCompact.Details>
+
+<StepHikeCompact.Code>
+
+```ts src/routes/api/auth/confirm/+server.ts
+import type { EmailOtpType } from '@supabase/supabase-js'
+import { redirect } from '@sveltejs/kit'
+
+import type { RequestHandler } from './$types'
+
+export const GET: RequestHandler = async ({ url, locals: { supabase } }) => {
+  const token_hash = url.searchParams.get('token_hash')
+  const type = url.searchParams.get('type') as EmailOtpType | null
+  const next = url.searchParams.get('next') ?? '/'
+
+  /**
+   * Clean up the redirect URL by deleting the Auth flow parameters.
+   *
+   * `next` is preserved for now, because it's needed in the error case.
+   */
+  const redirectTo = new URL(url)
+  redirectTo.pathname = next
+  redirectTo.searchParams.delete('token_hash')
+  redirectTo.searchParams.delete('type')
+
+  if (token_hash && type) {
+    const { error } = await supabase.auth.verifyOtp({ type, token_hash })
+    if (!error) {
+      redirectTo.searchParams.delete('next')
+      return redirect(307, redirectTo)
+    }
+  }
+
+  redirectTo.pathname = '/auth/error'
+  return redirect(307, redirectTo)
```

Comment:
/tangential

Curious about what the thought process was like around 303 vs 307. Did read [this page](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/307)  but still not too clear about when to opt for 307.

If you could share about the process of deciding between the two would be much appreciated

Reply:
good question, I think I may have made it 307 because I was looking at the sveltekit redirect docs when I wrote this, and I think in this case it has the same behavior as a 303. Going to change to a 303 for consistency with other parts of this page that should be 303